cmd/dump.go

```diff
@@ -0,0 +1,78 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cmd
+
+import (
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/elastic/elastic-package/internal/cobraext"
+	"github.com/elastic/elastic-package/internal/dump"
+	"github.com/elastic/elastic-package/internal/elasticsearch"
+)
+
+const dumpLongDescription = `Use this command as a exploratory tool to dump assets relevant for the package.`
+
+const dumpInstalledObjectsLongDescription = `Use this command to dump objects installed by Fleet as part of a package.
+
+Use this command as a exploratory tool to dump objects as they are installed by Fleet when installing a package. Dumped objects are stored in files as they are returned by APIs of the stack, without any processing.`
+
+func setupDumpCommand() *cobraext.Command {
+	dumpInstalledObjectsCmd := &cobra.Command{
+		Use:   "installed-objects",
+		Short: "Dump objects installed in the stack",
+		Long:  dumpInstalledObjectsLongDescription,
+		RunE:  dumpInstalledObjectsCmd,
+	}
+	dumpInstalledObjectsCmd.Flags().StringP(cobraext.DumpPackageFlagName, "p", "", cobraext.DumpPackageFlagDescription)
+	dumpInstalledObjectsCmd.MarkFlagRequired(cobraext.DumpPackageFlagName)
+	dumpInstalledObjectsCmd.Flags().StringP(cobraext.DumpOutputFlagName, "o", "output", cobraext.DumpOutputFlagDescription)
```

Comment:
Let's discuss possible `dump` use cases:

```
elastic-package dump installed-objects -p hello-world 
elastic-package dump assets -p hello-world
elastic-package dump assets -p hello-world 
elastic-package dump indices -p hello-world
elastic-package dump sample-event -p hello-world
```

WDYT about storing data in a structured way, for example `output`-dir can be replaced with `package-dump`, which may contain `installed-objects`, `indices`, etc.